
## ðŸ”— Projects

## 1. Gene Regulatory Network Inferences using Transformer-based models and scGPT

This project involved secondary data analysis based on studies by Cui et al. (2024) and Wang et al. (2024). All analyses were conducted on the Ilifu high-performance computing platform using python programming Language v3.13.2. The workflow adopted for our analysis is outlined in the steps below.

## Data Acquisition and Preprocessing

The Immune All Human scRNA-seq dataset was sourced from the GitHub repository (Cui et al., 2024 that contains the code used in inferring GRNs and subsequently pre-processed using the scGPT modelâ€™s internal pipeline, which consisted of three sequential steps. The initial step was filtering, which removed low-quality data by setting a minimum gene count threshold. Next, normalization was performed, where gene expression values were adjusted based on the total read count per cell to account for library size differences. The final step involved feature selection, where Highly Variable Genes (HVGs) were selected to retain the most biologically informative genes for subsequent analysis.

## scGPT-Based Gene Program Inference

## Model Initialization and Feature Extraction

The pretrained scGPT foundation model (Cui et al., 2024) was loaded into a Jupyter Notebook environment. This model incorporates a Gene Encoder for discrete gene identities, a Value Encoder for continuous expression levels, and a 12-layer Transformer Core responsible for learning complex gene-gene dependencies. Data-independent gene embeddings were extracted from the modelâ€™s representations. A filtering step was applied to select genes present in both the dataset's HVGs and the scGPT model's comprehensive vocabulary (>30,000 genes), resulting in a final set of 1,200 genes for embedding analysis.

## Clustering and Network Construction

The retrieved gene embeddings were used to construct a gene embedding network. Louvain clustering was performed on this network to partition genes into functional groups, or gene programs. To ensure biological relevance and stability, only clusters containing five or more genes were retained for subsequent analysis. A Cosine Similarity Network was then calculated between the identified gene programs, where edges represented the functional proximity of the programs, allowing for visualization of inter-program relationships.

## scGREAT-Based Consensus GRN Inference

The scGREAT pipeline, which adapts the Transformer architecture for GRN inference, was applied to the same Immune Human Dataset. The Biobert model was utilized to generate numerical embeddings for 12,303 unique gene names in batches of 128, resulting in a (12,303, 768)-dimensional embedding matrix.

## Sparse Network Generation and Correlation

An initial, sparse Gene Regulatory Network (GRN) was constructed using the k-Nearest Neighbours method, with k=50 to limit density. This generated an undirected graph of 12,303 nodes and 474,834 edges based on cosine similarity. Subsequently, an Expression Correlation Matrix was calculated using the Spearman method for the top 1,000 most active genes.

## Consensus Network Filtering

A Consensus GRN was generated by combining the sparse network with the expression correlation data. To refine the visualization and focus on high-confidence interactions, a rigorous filtering step was performed, retaining only the top 5% of all edges based on regulatory weight. This final network was used for identifying and visualizing the Top 200 central genes (hubs).

## Pathway and Functional Enrichment Analysis3.12

Pathway enrichment analysis was performed on the identified gene programs and regulatory hubs. Databases queried included Reactome, Gene Ontology (GO) Biological Process (GO: BP), and GO Molecular Function (GO:MF) to identify the known biological pathways and molecular roles associated with the gene set. Statistical significance was reported as the {-log} _ {10} ({Adjusted P-value}).

## 2. Leukemia Gene Expression Analysis(PCA, Variance Ranking & Exploratory Insights)

This project involved exploratory data analysis of leukemia gene expression data sourced from the Kaggle Database. This analysis was conducted on Visual Studio Code using python programming language v3.12.0. The workflow of the analysis is outlined below.

## Load data

The first few rows of the data was shown and the shape of the dataset in an interactive environment(Jupyter Notebooks).

## Exploratory Data Analysis

The information of the data was checked to show the memory usage and the columns available and the number of entries. A descriptive summary statistics to show the count, mean, standard deviation, upper and lower percentiles of the data was produced and the class balance was checked for the dataset.
Three classes were identified: Acute Lymphoblastic leukemia, Acute Myeloid leukemia and Healthy

## Missing values and duplicates

None were identified

## Distribution of gene expression

Matplotlib library was used to plot a histogram showing the distribution of the gene expression.

## Comparing expression across the diagnoses

Seaborn library was used to plot a boxplot showing the expression across Acute Myeloid Leukemia (AML), Acute Lymhoblastic Leukemia(ALL), and Healthy Diagnoses.

## Correlation between genes

This was tested by choosing the first 50 genes for readability. A heatmap showing the expression of the genes was produced and revealed the expression of the genes acroos the various samples within the dataset.
From Scikit-learn library, Principle Component Analysis(PCA) was used to reduce the dimensionality of the data and seaborn used to visualize a scatter plot.

## Feature variability

Variance for the top 20 genes was calculated to determine how variable the genes are, and the highly variable genes(HVGs) are most likely retained for further biological analysis.

## Key Findings 
Certain genes show distinct distribution patterns between diagnosis groups, suggesting potential biomarkers.
PC1 and PC2 explain a significant portion of variation, providing clear separation among samples.
The genes that showed high variance are top contributors or show strong influence on the various leukemia types.

Contact: Hmcmakhubela333@gmail.com


